subscription.js

```diff
@@ -45,7 +45,13 @@ class SubscriptionConnection {
     this.subscriber = subscriber
     this.subscriptionContexts = new Map()
 
-    this.socket.on('message', this.handleMessage.bind(this))
+    this.socket.on('message', (message) => {
+      try {
+        this.handleMessage(message)
+      } catch (e) {
+        this.handleConnectionClose()
+      }
```

Comment:
this should be a catch handler, because `handleMessage` returns a `Promise`. Can you add a test for this case? You might have to use https://www.npmjs.com/package/proxyquire to mock some things